common/types/chain.go

```diff
@@ -4,9 +4,10 @@ import "fmt"
 
 // Sequence represents the base type, for any chain's sequence object.
 // It should be convertible to a string
-type Sequence interface {
+type Sequence[SEQ any] interface {
 	fmt.Stringer
 	Int64() int64 // needed for numeric sequence confirmation - to be removed with confirmation logic generalization: https://smartcontract-it.atlassian.net/browse/BCI-860
+	Increment() SEQ // needed to produce the next usable sequence for a transaction
```

Comment:
This abstraction seems a little twisted up with the self reference - could that be avoided? Why do these need to increment themselves? And is increment a general enough name? Or are we really only interested in new/next nonces? Which may or may not depend on the last used once?

Reply:
Yeah it seems like passing a `func Next(last SEQ) SEQ` might be more flexible here :thinking:

Reply:
Is self reference a design pattern we want to avoid? I think passing the last sequence as a parameter would end up being used like `last.Next(last)` instead of just `last.Next()` which I think can accomplish the same thing

Reply:
> Is self reference a design pattern we want to avoid?

Yes, when not necessary, since it creates extra complexity. I also think it doesn't fit this model besides, because in some systems the sequence won't be dependent on the previous, and we might actually prefer to simplify this type in the future to make basic types usable, like big.Int.

More generally, requiring methods instead of funcs makes the API less flexible.

Reply:
Hope I'm understanding right but sounds like I should remove the `Increment` method from `Sequence` and instead pass a func like `type NextSequenceFunc[SEQ any] func (last SEQ) SEQ` to the Broadcaster as a parameter?

Reply:
I've made the changes I described above in the latest commit to make it easier to see